syntax = "proto3";

package communicator_objects;

enum ObservationTypeProto {
    DEFAULT = 0;
    GOAL_SIGNAL = 1;
}

message ObservationProto {
    message FloatData {
        repeated float data = 1;
    }

    repeated int32 shape = 1;
    oneof observation_data {
        FloatData float_data = 2;
    }
    repeated int32 dimension_properties = 3;
    string name = 4;
}
